### ISSUE TITLE
Problem with paths that have the same name as the API

### ISSUE THREAD

**Issue by janober** [Mashape/kong#458]
Hello,

came across kong and just wanted to play a little bit around with it. So started the docker container and simply added some apis and did redirect them some services I had running on localhost.

That worked fine for one test-api but I encountered an issue when I tried to serve html/image/javascript-files (know it is probably not best for that but the issue should be interesting anyway). It seemed to work for some files but for initially "random" seeming files I always got a 404. After some playing around I realized that it had to do with the name of the api. So all files/folders that started with the name of the api did not work.

My "test-api" was simply called "app". That resulted in having problems with files like this:

 - http://localhost:8000/app/scripts/app.js
 - http://localhost:8000/app/scripts/apple/test.js
 - http://localhost:8000/app/images/favicons/apple-touch-icon-114x114.png

So that is also not that great for real-apis because also they could run into the same issue.

The api got created like this:
```
curl -i -X POST \
 --url http://localhost:8001/apis/ \
 --data 'name=app' \
 --data 'target_url=http://localhost:8080/' \
 --data 'strip_path=true' \
 --data 'path=/app'
```

**Comment by thibaultCha** [comment]
The name of the API should have nothing to do with how the request is proxied. Could you please explain with more details what the issue is here? I don't see what is wrong exactly. What requests are failing? 

Because you have strip_path enabled, all requests under /app/xxx will be re routed to your upstream service at "/xxx".

**Comment by janober** [comment]
Exactly that is the issue that the api name seems to have to do with how the requests are proxied (or in my case not).

To take the example from above I would expect that it proxies like that:
```
http://localhost:8000/app/scripts/app.js
 -> http://localhost:8080/scripts/app.js

http://localhost:8000/app/scripts/apple/test.js
 -> http://localhost:8080/scripts/apple/test.js

http://localhost:8000/app/images/favicons/apple-touch-icon-114x114.png
 -> http://localhost:8080/images/favicons/apple-touch-icon-114x114.png
```

However for me that is not the case. For all of the above examples it get a 404. Even though the files exist. The issue is that all of the above examples have filename or foldernames that start with the app name. Because for all files/folder for which it is not the case it works fine. Here in bold to make it more obvious:
1. http://localhost:8000/app/scripts**/app**.js
2. http://localhost:8000/app/scripts**/app**le/test.js
3. http://localhost:8000/app/images/favicons**/app**le-touch-icon-114x114.png

For me it looks like if the "strip_path" is messing it up. When I change my local webserver that the "strip_path" is not needed and then create the route without it, it works fine.

So to make it clear this works:
```
http://localhost:8000/app/scripts/app.js
 -> http://localhost:8080/app/scripts/app.js

http://localhost:8000/app/scripts/apple/test.js
 -> http://localhost:8080/app/scripts/apple/test.js

http://localhost:8000/app/images/favicons/apple-touch-icon-114x114.png
 -> http://localhost:8080/app/images/favicons/apple-touch-icon-114x114.png
```

**Comment by thibaultCha** [comment]
Oh, I see now. This is not related to the name of the API but it is a bug with the path resolver, reported in #442.

**Comment by thibaultCha** [comment]
Just closed this via #465, thank you for the details.

**Comment by janober** [comment]
Great, thanks a lot!
